### ISSUE TITLE
WISH: .libPaths(new) to warn when it drops/ignore an item in `new`

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/Wishlist-for-R#85]
# Issue
`.libPaths(new)` may silently drop one or more entries in `new` if they don't exists.
For example,
```r
> .libPaths("non-existing")
> .libPaths()
[1] "/usr/local/lib/R/site-library" "/usr/lib/R/site-library"       "/usr/lib/R/library"           
```
And on Windows (only), it's picky about trailing slashes (which are not considered folders);
```r
## Ignored
> .libPaths("~/R/")
> .libPaths()
[1] "C:/Users/hb/Shared/R/x86_64-w64-mingw32-library/3.5"
[2] "C:/Program Files/R/R-3.5.1/library"

## OK
> .libPaths("~/R")
> .libPaths()
[1] "C:/Users/hb/Documents/R"                            
[2] "C:/Users/hb/Shared/R/x86_64-w64-mingw32-library/3.5"
[3] "C:/Program Files/R/R-3.5.1/library"
```

# Wish
Provide an informative warning when an entry is skipped, e.g.
```r
> .libPaths("non-existing")
Warning message:
No such folder: non-existing

> .libPaths("~/R/")
Warning message:
No such folder: ~/R/
```

For backward compatibility reason, one could add an argument `warn` defaulting to `warn = FALSE` for 100% backward compatibility or `warn = TRUE` for optional backward compatibility.

**Comment by rsettlage** [comment]
Add to this annoyance.  In cases where the the first path on the list is not writable, look to second etc.  For instance, in the below case, the first two paths are in a non-writable container, I do have permission to write to the last:
> .libPaths()
[1] "/usr/local/lib/R/site-library"
[2] "/usr/local/lib/R/library"
[3] "/home/rsettlag/R/OOD/Debian-9.6-3.5.1"

This will often give me an error when installing packages and the install script wants to update an existing package in first two locations which are not writable.

**Comment by HenrikBengtsson** [comment]
Curious though, why do you put (end up with?) "/home/rsettlag/R/OOD/Debian-9.6-3.5.1" at the back and not in the front?  The typical order should be:

1. User's package library folders (`R_LIBS_USER`)
2. Site-wide package library folders (`R_LIBS_SITE`)
3. System install package library folders (`R_LIBS`)

That way the user is in full control of what's found first - the way you have it now cause versions you install to be masked by even older versions of the same packages that are in the site-wide or system libraries.

**Comment by rsettlage** [comment]
Right, not a great setup.  This is totally unintentional and I am not sure what order is wrong.  This is an artifact of me trying to handle use cases on clusters where we are now allowing containers so we have potential version conflicts and OS conflicts.  The R code to handle OS isn't adequate imo, so I am trying to set all this for users and somehow ended up in the reverse order.  

Basically, I do this in the Dockerfile:
```
RUN sed -i '/^R_LIBS_USER=/d' /usr/local/lib/R/etc/Renviron
RUN echo 'R_ENVIRON=~/.Renviron.OOD \
      \nR_ENVIRON_USER=~/.Renviron.OOD \
      \n' >>/usr/local/lib/R/etc/Renviron
RUN rm /usr/local/lib/R/etc/Rprofile.site
```

Then I do this in my shell script to create the environ file (sorry about the Ruby):
```
singularity exec /groups/arcsingularity/<%= context.Rversion %> lsb_release -ir | awk '{ print $NF"-" }' | tr -d '\n' >>.Renviron.OOD.<%= context.Rversion %>
singularity exec /groups/arcsingularity/<%= context.Rversion %> Rscript -e 'cat(R.version$major,".",R.version$minor,sep="")' >>.Renviron.OOD.<%= context.Rversion %>
```

**Comment by HenrikBengtsson** [comment]
Hmm... I don't see how setting `R_LIBS_USER` and nothing else would cause this.  It should still appear on top.  If your talking about a Singularity container where `R_LIBS` and `R_LIBS_SITE` lives "in" the container and `R_LIBS_USER` outside in the users account on the host, all you should need to do is to set it to:
```
export R_LIBS_USER="~/R/%p-library/%v-Debian-9.6"
```
then R will automatically inject the architecture (`%p`) and R version (`%v`) on startup.  See `?.libPaths`. If not set, the default is `~/R/%p-library/%v`.  Also, don't use `.libPaths()` to set the package library path - just set `R_LIBS_USER` (and possibly `R_LIBS_SITE` but basically never `R_LIBS`).

**Comment by rsettlage** [comment]
Yes, both `R_LIBS` and `R_LIBS_SITE` live in the container and are not writable.  Outside the container, I have to deal with the possibility that users have an `.Renviron` file set up for normal system wide modules driven software, so that is why I am defining a special one.  I found that if they have this file, then the environment variable is ignored.  Additionally, I found cases where R started up and didn't translate the architecture/version variables and since I needed the OS version, I might as well get the rest.

I missed the first line:
```
echo "R_LIBS_USER= ${HOME}/R/OOD/" >.Renviron.OOD.<%= context.Rversion %>
singularity exec /groups/arcsingularity/<%= context.Rversion %> lsb_release -ir | awk '{ print $NF"-" }' | tr -d '\n' >>.Renviron.OOD.<%= context.Rversion %>
singularity exec /groups/arcsingularity/<%= context.Rversion %> Rscript -e 'cat(R.version$major,".",R.version$minor,sep="")' >>.Renviron.OOD.<%= context.Rversion %>
```

**Comment by HenrikBengtsson** [comment]
> Additionally, I found cases where R started up and didn't translate the architecture/version variables and since I needed the OS version, I might as well get the rest.

I never seen that happen.  If it does, then it's a bug and if you can put together a reproducible example, I'm pretty sure they're keen to fix it upstream.

**Comment by rsettlage** [comment]
I should also mention, this particular container is made for Rstudio, if I start Rstudio the order is as expected.  If I `singularity exec ... R` is is backwards.

```
> .libPaths()
[1] "/home/rsettlag/R/OOD/Debian-9.6-3.5.1" "/usr/local/lib/R/site-library"         "/usr/local/lib/R/library"  
```

Its a container, it is pretty reproducible.  Luckily, most people this is targeting are using the Rstudio interface so the reverse order bug is like me, N=1.
